### PR TITLE
LPS-104975 Open comment panel when clicking on comment button

### PR DESCRIPTION
**Pull request by julien**
Hey @wincent, 

I just wanted to ask you a review on this when you have time.
I didn't really understand how to adapt the previous `shouldActivate` functionality, but I'm not 100% sure that's an issue since it's all working.

Thanks

### CODE REVIEW

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/Sidebar.js [hunk 1]
```diff
@@ -79,57 +89,28 @@ export default function Sidebar() {
...
+			if (
+				plugin &&
+				typeof plugin.activate === 'function' &&
+				isMounted()
+			) {
+				plugin.activate();
+			} else if (!plugin) {
+				setHasError(true);
+			}
+		});
```
Comment (wincent) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/Sidebar.js]: It looks like there are two things happening here:

1. Via `getInstance`, if the current plugin (ie. as identified by `sidebarPanelId`) implements a `deactivate()` function, it calls it.
2. Via `registerPanel`, if the current plugin (ie. again, as identified by `sidebarPanelId`) implements an `activate()` function, it calls it.

This seems wrong. When you switched from one panel to another, the old code only called `deactivate` on the *previous* plugin, and `activate` on the *next* plugin. For example, if you switched from "fragments" to "widgets", "fragments" would receive a `deactivate()` call, then "widgets" would receive an `activate()` call; but now, "widgets" would receive a `deactivate()` then an `activate()`, and "fragments" won't receive anything at all.

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/actions/types.js [hunk 2]
```diff
@@ -20,5 +20,6 @@ export const MOVE_ITEM = 'MOVE_ITEM';
 export const PUBLISH = 'PUBLISH';
 export const REMOVE_ITEM = 'REMOVE_ITEM';
 export const UNLOAD_REDUCER = 'UNLOAD_REDUCER';
-export const UPDATE_LANGUAGE_ID = 'UPDATE_LANGUAGE_ID';
 export const UPDATE_ITEM_CONFIG = 'UPDATE_ITEM_CONFIG';
+export const UPDATE_LANGUAGE_ID = 'UPDATE_LANGUAGE_ID';
+export const UPDATE_SIDEBAR = 'UPDATE_SIDEBAR';
```
[the remark this answers is not in the document]
Comment (julien) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/actions/types.js]: :+1: Ok, I'll use `SWITCH_SIDEBAR_PANEL`

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/getSidebarPanelById.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
```
[the remark this answers is not in the document]
Comment (julien) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/getSidebarPanelById.js]: Yeah, I added a license header manually but didn't insert a blank line after it, which makes me think the second one was added automagically, but thanks for catching it, I hadn't seen that when sending the PR, :eyeglasses:

**Comment by wincent** [comment]
Going to close this because it is back in your queue now.

**Pull request closed without merging**